server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java

```diff
@@ -223,6 +224,10 @@ public Publication approvePublication(ResourceDescriptor resource) {
                 .filter(i -> i.getAction() == Publication.ResourceAction.ADD)
                 .toList();
 
+        List<Publication.Resource> resourcesToAddIfAbsent = publication.getResources().stream()
```

Comment:
Please check with checkReviewResources(resourcesToAddIfAbsent).

And delete deleteReviewResources(resourcesToAddIfAbsent).

Reply:
fixed